### PR TITLE
Expanding flower

### PR DESCRIPTION
**Pull request by Owlaukka**
Flower now starts off as a "small" widget in the top right corner and when you click the middle of it it expands wiht a simple animation. The icon circle is weird in the animation, but I didn't wanna waste too much time on that right now. No tests yet.

Changed the flower architecture quite a lot.

### CODE REVIEW

File: lib/screens/home/widgets/flower/petal_icon_button.dart
Comment (exeiya) [inline lib/screens/home/widgets/flower/petal_icon_button.dart]: What is this used for?

Comment (Owlaukka) [reply lib/screens/home/widgets/flower/petal_icon_button.dart]: I think it's very likely a leftover from when I was testing how different params affect the buttons and I just forgot to remove it. I'll make sure it's fine to delete it a bit later today.

If that's all the comments you have for this, can I merge this branch after this fix/check?

Comment (Owlaukka) [reply lib/screens/home/widgets/flower/petal_icon_button.dart]: Oh it's the color of when you are pressing the button. It's just quite small so it's prolly mostly covered by your finger. Still, it's a leftover from my experiements so I'll prolly delete it.

Comment (exeiya) [reply lib/screens/home/widgets/flower/petal_icon_button.dart]: Okay, I was just wondering if we want to leave it like that or not.

**Review by exeiya — APPROVED** [review]
Everything seems to work as it should at this stage, though we probably need to figure out how to get the animation working smoothly for the icons too. That can be done later though so that's not an issue right now. 
Also I was thinking that the box shadow the flower has could maybe have a different color or it could be a little bit lighter shade of green. 

Anyway, no issues that need a fix right now so I'll merge this branch to pre_release and after that we can merge pre_release to the master branch.

**Pull request merged**
